Kitodo/src/main/java/org/kitodo/production/interfaces/activemq/CreateNewProcessesProcessor.java

```diff
@@ -0,0 +1,220 @@
+            InvalidMetadataValueException, IOException, NoRecordFoundException, NoSuchMetadataFieldException,
+            ParserConfigurationException, ProcessGenerationException, ProcessorException, SAXException,
+            TransformerException, UnsupportedFormatException, URISyntaxException, XPathExpressionException {
+
+        List<TempProcess> processHierarchy = importService.importProcessHierarchy(
+                order.getImports().get(which).getValue(), order.getImports().get(which).getKey(),
+                order.getProjectId(), order.getTemplateId(), IMPORT_WITHOUT_ANY_HIERARCHY,
+                rulesetManagement.getFunctionalKeys(FunctionalMetadata.HIGHERLEVEL_IDENTIFIER));
+        if (processHierarchy.size() == 0) {
+            throw new ProcessorException("Process was not imported");
+        } else if (processHierarchy.size() > 1) {
+            throw new ProcessorException(processHierarchy.size() + " processes were imported");
+        }
+        return processHierarchy.get(0);
+    }
+
+    /* In the second and middle part of the processing routine, the process
+     * title is generated. */
+    private Process formProcessTitle(CreateNewProcessOrder order, TempProcess tempProcess, Process process)
+            throws DAOException, ProcessGenerationException, ProcessorException {
+
+        ProcessFieldedMetadata processDetails = ProcessHelper.initializeProcessDetails(tempProcess.getWorkpiece()
+                .getLogicalStructure(), rulesetManagement, ACQUISITION_STAGE_PROCESS_CREATION, METADATA_LANGUAGE);
+        Process parentProcess = order.getParent();
+        ProcessHelper.generateAtstslFields(tempProcess, processDetails.getRows(), Collections.emptyList(), tempProcess
+                .getWorkpiece().getLogicalStructure().getType(), rulesetManagement, ACQUISITION_STAGE_PROCESS_CREATION,
+            METADATA_LANGUAGE, parentProcess, true);
+        if (order.getTitle().isPresent()) {
+            process.setTitle(order.getTitle().get());
+        }
+        if (!ProcessValidator.isProcessTitleCorrect(process.getTitle())) {
+            throw new ProcessorException(Helper.getTranslation("processTitleAlreadyInUse", process.getTitle()));
+        }
+        return parentProcess;
+    }
+
+    /* In the third and final part of the processing routine, the process is
+     * created and saved. */
+    private void createProcess(TempProcess tempProcess, Process process, Process parentProcess) throws DataException,
+            CustomResponseException, IOException, CommandException {
+
+        saveProcess(process);
+        fileService.createProcessLocation(process);
+        metsService.saveWorkpiece(tempProcess.getWorkpiece(), processService.getMetadataFileUri(process));
+        if (Objects.nonNull(parentProcess)) {
+            MetadataEditor.addLink(parentProcess, LAST_CHILD, process.getId());
+            process.setParent(parentProcess);
```

Comment:
The parent process is only set if we provide a explicit parentID in the ActiveMQ message. Not if there is a process in Kitodo whose `recordIdentifier` is referenced by the `higherlevelIdentifier`.